force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +161,22 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        List<Opportunity> newOpps = new List<Opportunity>();
+        Account newAcc = new Account();
+        newAcc.Name = accountName;
+        insert newAcc;
         // Loop through each Opportunity name
+        for (String oppName : oppNames) {
             // Create an Opportunity with the given name
+            Opportunity newOpp = new Opportunity();
+            newOpp.Name = oppName;
+            newOpp.AccountId = newAcc.Id;
+            newOpp.StageName = 'Qualification';
+            newOpp.CloseDate = Date.newInstance(2024, 3, 27);
             // Add the Opportunity to the list
+            newOpps.add(newOpp);
+        }
+        upsert newOpps;
```

Comment:
Remember we first want to see if there are any opps with the names in the oppNames argument and update those.  This is a tricky one